### PR TITLE
Add jobs abstractions

### PR DESCRIPTION
**Pull request by collijk**
### Description

- *Category*: refactor
- *JIRA issue*:  [MIC-2881](https://jira.ihme.washington.edu/browse/MIC-2881) 

Extract some utilities for managing jobs and clean up configuration of workers.

### Testing
CI

### CODE REVIEW

File: src/vivarium_cluster_tools/psimulate/branches.py [hunk 1]
```diff
@@ -100,6 +100,8 @@ def __iter__(self):
         for job_config in product(
             self._keyspace["input_draw"], self._keyspace["random_seed"], self.branches
         ):
+            if job_config[2] is None:
```
Comment (collijk) [inline src/vivarium_cluster_tools/psimulate/branches.py:103]: If there are no branches, this now returns an empty dict instead of None, so we're dealing with consistent types.

File: src/vivarium_cluster_tools/psimulate/distributed_worker.py [hunk 2]
```diff
@@ -98,66 +98,42 @@ def fork_work_horse(self, job, queue):
...
 
-    input_draw = parameters["input_draw"]
-    random_seed = parameters["random_seed"]
-    model_specification_file = parameters["model_specification_file"]
-    branch_config = parameters["branch_configuration"]
+    job_parameters = jobs.JobParameters(**job_parameters)
+
     logger.info(f"Launching new job {job} on {node}")
-    logger.info(
-        "Starting job: {}".format(
```
Comment (collijk) [inline src/vivarium_cluster_tools/psimulate/distributed_worker.py:111]: See JobParameters.__repr__

File: src/vivarium_cluster_tools/psimulate/distributed_worker.py [hunk 3]
```diff
@@ -98,66 +98,42 @@ def fork_work_horse(self, job, queue):
...
-            (input_draw, random_seed, model_specification_file, branch_config)
-        )
-    )
+    logger.info(f"Starting job: {job_parameters}")
 
     try:
-        np.random.seed([input_draw, random_seed])
```
Comment (collijk) [inline src/vivarium_cluster_tools/psimulate/distributed_worker.py:117]: Totally irrelevant seed setting.

File: src/vivarium_cluster_tools/psimulate/distributed_worker.py [hunk 4]
```diff
@@ -98,66 +98,42 @@ def fork_work_horse(self, job, queue):
...
-        run_key = {"input_draw": input_draw, "random_seed": random_seed}
-
-        if branch_config is not None:
-            configuration.update(dict(branch_config))
-            run_key.update(dict(branch_config))
-
-        input_data_config = {
-            "input_draw_number": input_draw,
-        }
-        if vct_globals.ARTIFACT_PATH_KEY in dict(branch_config).get(
```
Comment (collijk) [inline src/vivarium_cluster_tools/psimulate/distributed_worker.py:133]: This is some baffling logic.  We've already put the branch_configuration in the configuration and if it contains the key input_data.artifact_path, we're going to overwrite it with the same thing again with this + line 156 (old code)

File: src/vivarium_cluster_tools/psimulate/distributed_worker.py [hunk 5]
```diff
@@ -98,66 +98,42 @@ def fork_work_horse(self, job, queue):
...
+        configuration = job_parameters.branch_configuration
+        # TODO: Need to test serialization of an empty dict, then this
+        #   can go away.  If you're successfully running code and this
+        #   assert is still here, delete it.
+        assert configuration is not None
 
         configuration.update(
             {
                 "run_configuration": {
-                    "input_draw_number": input_draw,
```
Comment (collijk) [inline src/vivarium_cluster_tools/psimulate/distributed_worker.py:147]: input draw is already in the run key, no need to duplicate.

File: src/vivarium_cluster_tools/psimulate/distributed_worker.py [hunk 6]
```diff
@@ -206,13 +182,14 @@ def worker(parameters: Mapping):
...
+        do_sim_epilogue(start_snapshot, end_snapshot, event, exec_time, job_parameters)
 
         idx = pd.MultiIndex.from_tuples(
-            [(input_draw, random_seed)], names=["input_draw_number", "random_seed"]
+            [(job_parameters.input_draw, job_parameters.random_seed)],
+            names=["input_draw_number", "random_seed"],
         )
         output_metrics = pd.DataFrame(metrics, index=idx)
-        for k, v in collapse_nested_dict(run_key):
+        for k, v in collapse_nested_dict(job_parameters.branch_configuration):
```
Comment (collijk) [inline src/vivarium_cluster_tools/psimulate/distributed_worker.py:192]: This will break results processing, which has to drop these columns, but that should be a one line fix.

**Review by rmudambi — APPROVED** [review]
